Tests/MaterialExtractTest.cs

```diff
@@ -71,5 +72,53 @@ public void TestVmatExtract()
 
             Assert.That(materialExtract.ToValveMaterial(), Is.Not.Empty);
         }
+
+        public static IEnumerable<TestCaseData> PngImageChannelsSource()
+        {
+            var c1234 = new SKColor(1, 2, 3, 4);
+            yield return new TestCaseData(c1234, ChannelMapping.R, new SKColor(1, 1, 1));
+            yield return new TestCaseData(c1234, ChannelMapping.G, new SKColor(2, 2, 2));
+            yield return new TestCaseData(c1234, ChannelMapping.B, new SKColor(3, 3, 3));
+            yield return new TestCaseData(c1234, ChannelMapping.A, new SKColor(4, 4, 4, 255));
+
+            yield return new TestCaseData(c1234, ChannelMapping.RG, new SKColor(1, 2, 3, 255));
+            yield return new TestCaseData(c1234, ChannelMapping.RGB, new SKColor(1, 2, 3, 255));
+
+            yield return new TestCaseData(c1234, ChannelMapping.AG, new SKColor(4, 2, 0));
+
+            // Alpha is 4, so it multiplies the colors making them 0
+            // should be c1234
+            yield return new TestCaseData(c1234, ChannelMapping.RGBA, new SKColor(0, 0, 0, 4));
```

Comment:
Skia png encode is removing color data based on alpha value, which is not what we want to happen. A consistent repro is exporting any vmorf texture.